src/hotspot/share/cds/classListParser.cpp

```diff
@@ -458,24 +466,22 @@ InstanceKlass* ClassListParser::load_class_from_source(Symbol* class_name, TRAPS
     THROW_NULL(vmSymbols::java_lang_ClassNotFoundException());
   }
 
-  InstanceKlass* k = ClassLoaderExt::load_class(class_name, _source, CHECK_NULL);
+  InstanceKlass* k = UnregisteredClasses::load_class(class_name, _source, CHECK_NULL);
   if (k->local_interfaces()->length() != _interfaces->length()) {
     print_specified_interfaces();
     print_actual_interfaces(k);
     error("The number of interfaces (%d) specified in class list does not match the class file (%d)",
           _interfaces->length(), k->local_interfaces()->length());
   }
 
+  k->clear_shared_class_loader_type();
```

Comment:
I think this may not be necessary, as the correct loader type should have been set inside  ClassLoaderExt::record_result(). Maybe change this to?

```
assert(k->is_shared_unregistered_class(), "must be");
```

Reply:
I've changed it to assert. Also removed the `clear_shared_class_loader_type()` since it is not used anymore.
The class loader type is not set not due to the call to `ClassLoaderExt::record_result()` which will default the loader type to boot loader. For unregistered classes, the `ClassLoaderExt::record_result()` won't be called in `ClassLoader::record_result()`.